lib/addons/appmesh/index.ts

```diff
@@ -13,13 +50,19 @@ export class AppMeshAddOn implements ClusterAddOn {
         const sa = cluster.addServiceAccount('appmesh-controller', opts);
 
         // Cloud Map Full Access policy.
-        const cloudMapPolicy = ManagedPolicy.fromAwsManagedPolicyName("AWSCloudMapFullAccess")
+        const cloudMapPolicy = ManagedPolicy.fromAwsManagedPolicyName("AWSCloudMapFullAccess");
         sa.role.addManagedPolicy(cloudMapPolicy);
 
         // App Mesh Full Access policy.
-        const appMeshPolicy = ManagedPolicy.fromAwsManagedPolicyName("AWSAppMeshFullAccess")
+        const appMeshPolicy = ManagedPolicy.fromAwsManagedPolicyName("AWSAppMeshFullAccess");
         sa.role.addManagedPolicy(appMeshPolicy);
 
+        if(this.appMeshOptions.enableTracing && this.appMeshOptions.tracingProvider === "x-ray") {
```

Comment:
x-ray an enum...then you can switch on these values.

Reply:
I added type restrictions which is less verbose